### PR TITLE
Upgraded Azure Pipelines to support demands

### PR DESCRIPTION
**Pull request by Eldarrin**
Signed-off-by: mortx <mortxbox@live.com>

Old PR: https://github.com/kedacore/keda/pull/2707

You can now supply a master template to support demands. NOTE: You must have a master supporting agent even if it is offline!

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Relates to https://github.com/kedacore/keda-docs/pull/685
Relates to https://github.com/kedacore/keda/issues/2328

### CODE REVIEW

**Comment by tomkerkhove** [comment]
@JorTurFer would you mind reviewing this one when you have time? It's been open for a while

**Comment by JorTurFer** [comment]
I'm waiting your opinions is docs

**Comment by tomkerkhove** [comment]
@Eldarrin Any thought if we can close this PR soon? We are planning to release in 2 weeks.

**Comment by Eldarrin** [comment]
@tomkerkhove Its all fine and ready to be merged I believe. I'm actually running it in production at my client's organisation.

**Comment by tomkerkhove** [comment]
Nice to hear! But we should probably move to the approach proposed by @JorTurFer if that's OK for you?

**Comment by Eldarrin** [comment]
Ok, let me have a look. Want me to take a new fork and start from there?

**Comment by tomkerkhove** [comment]
Thank you @JorTurFer Do you want to have a look?

**Review by JorTurFer — APPROVED** [review]
Looking good! ❤️

**Comment by JorTurFer** [comment]
I understand the "parent" approach, but requiring an already deployed deployment only for registering the job looks like overkill. Nothing to add to the code itself, but I think that we need to solve the approach for that in the docs repo.

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 - **General:** `external` extension reduces connection establishment with long links ([#3193](https://github.com/kedacore/keda/issues/3193))
 - **General:** Use `mili` scale for the returned metrics ([#3135](https://github.com/kedacore/keda/issue/3135))
 - **General:** Use more readable timestamps in KEDA Operator logs ([#3066](https://github.com/kedacore/keda/issue/3066))
+- **Azure Pipelines Scaler:** Add support for Azure Pipelines to support demands (capabilities) ([#2795](https://github.com/kedacore/keda/pull/2795))
```
[the remark this answers is not in the document]
Comment (Eldarrin) [reply CHANGELOG.md]: done, I stupidly thought P comes before S. doh

**Review by JorTurFer — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@
 ### Improvements
 
 - **Azure Event Hub Scaler:** Improve logging in Event hubs scaler when blob container not found ([#2363]https://github.com/kedacore/keda/issues/2363)
+- **Azure Pipelines:** Added the ability to support ADO demands by referencing a master template ([#2707](https://github.com/kedacore/keda/pull/2707))
```
[the remark this answers is not in the document]
Comment (Eldarrin) [reply CHANGELOG.md]: Hi @zroubalik I think you are looking at the old PR which was closed, that doesn't match my commit. I will update mine to issue not pull though

**Comment by JorTurFer** [comment]
@Eldarrin 
We have plan to do a release aiming to next week, could you solve the conflicts in `pkg/scalers/azure_pipelines_scaler.go` to merge this PR?

**Comment by JorTurFer** [comment]
There is a [code linting issue](https://github.com/kedacore/keda/runs/7576361526?check_suite_focus=true), could you solve it please?

**Comment by zroubalik** [comment]
@JorTurFer @v-shenoy @tomkerkhove what is the status of this?

**Comment by JorTurFer** [comment]
This PR is ready but we are waiting till some things are solved in docs

**Pull request merged**